bika/lims/upgrade/v01_01_003.py

```diff
@@ -26,10 +26,37 @@ def upgrade(tool):
 
     logger.info("Upgrading {0}: {1} -> {2}".format(product, ver_from, version))
 
+    # Updating profile steps:
+    # list of the generic setup import step names:
+    # --> portal.portal_setup.getSortedImportSteps() <---
+    #
+    # if you want more metadata use this:
+    # --> portal.portal_setup.getImportStepMetadata('jsregistry') <---
+    #
+    # important info about upgrade steps in
+    # http://stackoverflow.com/questions/7821498/is-there-a-good-reference-list-for-the-names-of-the-genericsetup-import-steps
+
+    # New ARImports workflow
+    setup.runImportStepFromProfile(profile, 'workflow')
+
     # Attachments must be present in a catalog, otherwise the idserver
     # will fall apart.  https://github.com/senaite/bika.lims/issues/323
     at = getToolByName(portal, 'archetype_tool')
     at.setCatalogsByType('Attachment', ['portal_catalog'])
-
+    update_arimport_workflow_and_permissions(portal, setup)
     logger.info("{0} upgraded to version {1}".format(product, version))
     return True
+
+
+def update_arimport_workflow_and_permissions(portal, setup):
+    """
+    Updates the workflow and permissions of ARImprt objects since a new
+    workflow has been created for them.
+    :param portal: Portal archetype object
+    :return: None
+    """
+    wf_tool = getToolByName(portal, 'portal_workflow')
+    logger.info("Updating workflow definitions...")
+    wf = wf_tool.getWorkflowById("bika_arimports_workflow")
+    wf.updateRoleMappingsFor(setup.arimports)
+    logger.info("Workflow definitions updated.")
```

Comment:
Since permission mappings for `bika_arimport_workflow` were changed in https://github.com/senaite/bika.lims/commit/6a6b58c1a9a079ffdfb1152aedfc002db24ae6f3 , shouldn't be also necessary an `updateRoleMappingsFor`, but for each `ARImport` preexisting object?

Reply:
Yes, it is